src/Models/HydrostaticFreeSurfaceModels/update_hydrostatic_free_surface_model_state.jl

```diff
@@ -68,18 +197,24 @@ end
 
 function compute_auxiliaries!(model::HydrostaticFreeSurfaceModel; w_parameters = tuple(w_kernel_parameters(model.grid)),
                                                                   p_parameters = tuple(p_kernel_parameters(model.grid)),
-                                                                  κ_parameters = tuple(:xyz)) 
-    
+                                                                  κ_parameters = tuple(:xyz))
+
     grid = model.grid
     closure = model.closure
     diffusivity = model.diffusivity_fields
 
     for (wpar, ppar, κpar) in zip(w_parameters, p_parameters, κ_parameters)
         compute_w_from_continuity!(model; parameters = wpar)
+
         compute_diffusivities!(diffusivity, closure, model; parameters = κpar)
-        update_hydrostatic_pressure!(model.pressure.pHY′, architecture(grid), 
-                                    grid, model.buoyancy, model.tracers; 
-                                    parameters = ppar)
+
+        update_hydrostatic_pressure!(model.pressure.pHY′, architecture(grid),
+                                     grid, model.buoyancy, model.tracers;
+                                     parameters = ppar)
     end
+
     return nothing
 end
+
+# TO DELETE!!!!! (We aim to do single pass)
+second_pass_of_fill_halo_regions!(grid, velocities, args...) = nothing
```

Comment:
is this needed for the grid coord/metric halo filling?

Reply:
I guess we can remove this if it's not used (it doesn't seem like it is used in the above `update_state!`

Reply:
Though not required for filling the halo regions of grid coordinates/metrics, its functionality is imported in src/MultiRegion/multi_region_cubed_sphere_grid.jl and extended for a conformal cubed sphere grid.